### PR TITLE
feat: use two buffer pointers for init and media segments separately

### PR DESCRIPTION
**Pull request by lfordyce**

### CODE REVIEW

**Review by chenda6 — APPROVED** [review]

File: packager/live_packager.cc [hunk 1]
```diff
@@ -80,6 +80,39 @@ class SegmentDataReader {
...
+      memcpy(buffer, media_segment_.Data() + position_, second_chunk_size);
+      position_ += second_chunk_size;
+      return second_chunk_size;
+    }
+  }
+
+ private:
+  const Segment& init_segment_;
+  const Segment& media_segment_;
+  uint64_t init_position_ = 0;
```
Comment (Vadym-Kupriyanchuk) [inline packager/live_packager.cc:112]: I think you can implement this functionality with only one position. (but maybe I am missing something)
```
if (pos < init.size()) {
 // return data from the init segment
}

if(pos > init.size()) {
segPos = pos - init.size()
//return data from  media segment with segPos
}

```

File: packager/live_packager_test.cc [hunk 2]
```diff
@@ -168,9 +168,9 @@ TEST_F(LivePackagerBaseTest, EncryptionFailure) {
     std::vector<uint8_t> segment_buffer = ReadTestDataFile(segment_num);
     ASSERT_FALSE(segment_buffer.empty());
 
-    FullSegmentBuffer in;
-    in.SetInitSegment(init_segment_buffer.data(), init_segment_buffer.size());
-    in.AppendData(segment_buffer.data(), segment_buffer.size());
+    SegmentData init_seg(init_segment_buffer.data(),
```
Comment (Vadym-Kupriyanchuk) [inline packager/live_packager_test.cc:171]: **nit**: finally need to increase line width for clang-format,  it's annoying to see all these wraps (((

Comment (lfordyce) [reply packager/live_packager_test.cc:171]: Initially I was just following the Contributing Guid for clang-format: 
https://github.com/Pluto-tv/shaka-packager-live/blob/pluto-cmake/docs/source/build_instructions.md#contributing. 

Do we want to change or modify this? Maybe `ColumnLimit: 100`?

Comment (chenda6) [reply packager/live_packager_test.cc:171]: This is certainly debatable, but I'd respect the standard of the existing codebase and keep this as is.  Just my 2 cents.  We can take a vote on this too.

**Review by chenda6 — DISMISSED** [review]

File: packager/live_packager.cc [hunk 3]
```diff
@@ -80,6 +80,40 @@ class SegmentDataReader {
...
+
+  uint64_t Read(void* buffer, uint64_t size) {
+    if (position_ < init_segment_.Size()) {
+      const uint64_t first_chunk_size =
+          std::min(size, init_segment_.Size() - position_);
+      memcpy(buffer, init_segment_.Data() + position_, first_chunk_size);
+
+      position_ += first_chunk_size;
+      return first_chunk_size;
+    } else {
```
Comment (Vadym-Kupriyanchuk) [inline packager/live_packager.cc:97]: you don't need **else** here, previous **if** always finishes with a return

File: packager/live_packager_test.cc (hunk 2 above)
Comment (Vadym-Kupriyanchuk) [reply packager/live_packager_test.cc:171]: Existing settings make me think about what type of computer shaka developers are using. Right now on a laptop, I have ~70% of free horizontal space with their formatting.

**Review by Vadym-Kupriyanchuk — DISMISSED** [review]

**Review by chenda6 — APPROVED** [review]

File: packager/live_packager_test.cc (hunk 2 above)
Comment (chenda6) [reply packager/live_packager_test.cc:171]: This is outside the scope of this PR.  Let's have a follow-up discussion and address this in a separate PR.

**Pull request merged**